syntax/compile_xstr.go

```diff
@@ -156,3 +158,53 @@ func xstrConcat(seq rel.Value) (rel.Value, error) {
 	}
 	return rel.NewString([]rune(sb.String())), nil
 }
+
+// this function cleans whitespaces of bare strings before and after a computed emptyt string
+func cleanEmptyVal(values rel.Value) []rel.Value {
```

Comment:
Nit: If the input must be a rel.Array, I'd be inclined to make the parameter a rel.Array. Otherwise you're just hiding information for no benefit.